src/sidebar/services/groups.js

```diff
@@ -348,25 +323,103 @@ export default function groups(
       directLinkedGroupId
     );
 
+    addGroupsToStore(groups, [
+      directLinkedAnnotationGroupId,
+      directLinkedGroupId,
+      store.getDefault('focusedGroup'),
+    ]);
+
+    return groups;
+  }
+
+  /**
+   * Load the specific groups configured by the annotation service.
+   *
+   * @param {string[]} groupIds
+   */
+  async function loadServiceSpecifiedGroups(groupIds) {
+    let error;
+    const tryFetchGroup = async id => {
+      try {
+        return await fetchGroup(id);
+      } catch (e) {
+        error = e;
+        return null;
+      }
+    };
+
+    const groups = (
+      await Promise.all(groupIds.map(id => tryFetchGroup(id)))
+    ).filter(g => g !== null);
+
+    addGroupsToStore(groups);
+
+    if (error) {
+      toastMessenger.error(`Unable to fetch groups: ${error.message}`, {
+        autoDismiss: false,
+      });
+    }
+
+    return groups;
+  }
+
+  /**
+   * @param {Group[]} groups
+   * @param {string[]} groupsToFocus - IDs of groups to focus, in preference order
+   */
+  function addGroupsToStore(groups, groupsToFocus = []) {
```

Comment:
The `groupsToFocus` parameter gives me a touch of confusion: we only ever focus one group. What's the background of taking an Array here? Is it in case one or more of the groups indicated here hasn't actually been fetched? Perhaps a little more commenting?